File: src/main/java/me/foodbag/hello/service/FoodServiceImpl.java
package me.foodbag.hello.service;

import me.foodbag.hello.persistence.model.Food;
import me.foodbag.hello.persistence.repository.FoodRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class FoodServiceImpl implements FoodService {

  @Autowired private FoodRepository foodRepository;

  @Override
  public Food getFoodById(Long id) {
    return foodRepository.findById(id).orElse(null);
  }

  @Override
  public Food getFoodByName(String name) {
    return foodRepository.findByName(name);
  }

  @Override
  public List<Food> getAllFoods() {
    return foodRepository.findAll();
  }

  @Override
  public boolean exists(String name) {
    return foodRepository.findByName(name) != null;
  }

  @Override
  public Food save(Food food) {
    return foodRepository.save(food);
  }
}

File: src/main/java/me/foodbag/hello/service/FoodServiceImpl.kt
package me.foodbag.hello.service

import me.foodbag.hello.persistence.model.Food
import me.foodbag.hello.persistence.repository.FoodRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
open class FoodServiceImpl : FoodService {

    @Autowired
    private val foodRepository: FoodRepository? = null

    override val allFoods: List<Food>
        get() = foodRepository!!.findAll()

    override fun getFoodById(id: Long?): Food {
        return foodRepository!!.findById(id!!).orElse(null)
    }

    override fun getFoodByName(name: String): Food {
        return foodRepository!!.findByName(name)
    }

    override fun exists(name: String): Boolean {
        return foodRepository!!.findByName(name) != null
    }

    override fun save(food: Food): Food {
        return foodRepository!!.save(food)
    }
}
